### ISSUE TITLE
【从零开始的 NLP：使用字符级 RNN 分类名称】实例代码错误

### ISSUE THREAD

**Issue by yoomy** [apachecn/pytorch-doc-zh#561]
创建网络

import torch.nn as nn

class RNN(nn.Module):
    def __init__(self, input_size, hidden_size, output_size):
        super(RNN, self).__init__()

        self.hidden_size = hidden_size

        self.i2h = nn.Linear(input_size + hidden_size, hidden_size)
        self.i2o = nn.Linear(input_size + hidden_size, output_size)
        self.softmax = nn.LogSoftmax(dim=1)

    def forward(self, input, hidden):
        combined = torch.cat((input, hidden), 1)
        hidden = self.i2h(combined)
        output = self.i2o(combined)
        output = self.softmax(output)
        return output, hidden

    def initHidden(self):
        return torch.zeros(1, self.hidden_size)

n_hidden = 128
rnn = RNN(n_letters, n_hidden, n_categories)

错误位置：output = self.i2o(combined)
修正：output = self.i2o(hidden)

**Comment by jiangzhonglian** [comment]
首先感谢你的好意，提出问题，不过这个没错。

你看看 size，如果传入hidden 会报错（你再理解，返回值的意义）
self.i2o = nn.Linear(input_size + hidden_size, output_size)

还有,你看官方网站： https://pytorch.org/tutorials/intermediate/char_rnn_classification_tutorial.html

<img width="958" alt="image" src="https://user-images.githubusercontent.com/9199175/127593407-4d9c2193-11ff-4ef5-ad12-def49479035f.png">
